Userland/Libraries/LibWeb/Animations/Animation.cpp

```diff
@@ -23,13 +23,15 @@ JS_DEFINE_ALLOCATOR(Animation);
 // https://www.w3.org/TR/web-animations-1/#dom-animation-animation
 JS::NonnullGCPtr<Animation> Animation::create(JS::Realm& realm, JS::GCPtr<AnimationEffect> effect, JS::GCPtr<AnimationTimeline> timeline)
 {
+    auto& vm = realm.vm();
+
     // 1. Let animation be a new Animation object.
     auto animation = realm.heap().allocate<Animation>(realm, realm);
 
     // 2. Run the procedure to set the timeline of an animation on animation passing timeline as the new timeline or, if
     //    a timeline argument is missing, passing the default document timeline of the Document associated with the
     //    Window that is the current global object.
-    if (!timeline) {
+    if (vm.argument_count() < 2) {
```

Comment:
hrm. feels awkward to reach for vm.argument count here. Would be nice if we could tell the IDLGenerator in the future that a missing argument is significant for the call, instead of it papering it over by passing a defaulted argument value.

Reply:
Agreed, I thought the same thing while writing it. Made #23220, and I'll probably take a look at it this weekend